core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -431,6 +435,26 @@ private SqlNode getIdentifier(SqlBasicCall intervalOperand) {
     return ((SqlBasicCall) aggCall).operand(0);
   }
 
+  private void unparseTimestamp(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    if (call.getOperator().toString().equals("TIMESTAMP_ADD")
+            || call.getOperator().toString().equals("TIMESTAMP_SUB")) {
+      final SqlWriter.Frame formatTime = writer.startFunCall("TIMESTAMPADD");
+      String formatCall = ((SqlIntervalLiteral) call.operand(1))
+              .getTypeName().toString().split("_")[1];
+      writer.print(formatCall + ", ");
+      String intCall = ((SqlIntervalLiteral) call.operand(1)).getValue().toString();
+      if (call.getOperator().toString().equals("TIMESTAMP_SUB")) {
+        writer.print("-" + intCall + ", ");
```

Comment:
`if (true) {
writer.print("-"); 
}
writer.print(intCall + ", ")
`
and remove else part